src/extension/security/cas/src/main/java/org/geoserver/security/cas/GeoServerCasAuthenticationFilter.java

```diff
@@ -258,4 +270,63 @@ public void logout(
             Authentication authentication) {
         request.setAttribute(GeoServerLogoutFilter.LOGOUT_REDIRECT_ATTR, casLogoutURL);
     }
+
+    @Override
+    protected Collection<GeoServerRole> getRoles(HttpServletRequest request, String principal)
```

Comment:
Not really and issue, but for completeness I guess that would be good to complete the principal roles list by using the RoleCalculator

```
RoleCalculator calc =
                        new RoleCalculator(getSecurityManager().getActiveRoleService());
```

In order to be sure we add all the roles associated to the current user accordingly to the active role service too.

Reply:
Interesting, I had no idea this class existed. The base class does not do anything with the RoleCalculator, but maybe it should not be done when picking roles from a specific role service or a specific user group service. 
Reading from the headers does nothing either, but I see the J2EE auth (another subclass of the pre authenticated filter) using it.... adding it in the "roles from custom attributes" path only, don't want to touch the base classes.